src/main/java/com/rtb/projectmanagementtool/task/TaskData.java

```diff
@@ -39,35 +38,36 @@ public TaskData(
   }
 
   public TaskData(Entity entity) {
-    taskID = (long) entity.getKey().getId();
+    taskID = (long) entity.getProperty("taskID");
     projectID = (long) entity.getProperty("projectID");
     name = (String) entity.getProperty("name");
     description = (String) entity.getProperty("description");
     status = Status.valueOf((String) entity.getProperty("status"));
-    try {
-      users = new HashSet<Long>((ArrayList<Long>) entity.getProperty("users"));
-    } catch (ClassCastException e) {
-      users = (HashSet<Long>) entity.getProperty("users");
-    } catch (NullPointerException e) {
-      users = new HashSet<Long>();
+    if (entity.getProperty("users") == null) {
+      users = new ArrayList<Long>();
+    } else {
+      users = (ArrayList<Long>) entity.getProperty("users");
     }
-    try {
-      subtasks = new HashSet<Long>((ArrayList<Long>) entity.getProperty("subtasks"));
-    } catch (ClassCastException e) {
-      subtasks = (HashSet<Long>) entity.getProperty("subtasks");
-    } catch (NullPointerException e) {
-      subtasks = new HashSet<Long>();
+    if (entity.getProperty("subtasks") == null) {
+      subtasks = new ArrayList<Long>();
+    } else {
+      subtasks = (ArrayList<Long>) entity.getProperty("subtasks");
     }
   }
 
   public Entity toEntity() {
     Entity entity = new Entity("Task", taskID);
+    entity.setProperty("taskID", taskID);
```

Comment:
What's the plan for generating taskID when inserting new records into datastore?

Reply:
It's probably not the best plan, but in the server I'll be creating an entity without the id parameter, so the entity will auto-generate an id. Then, I'll get the id from the entity and use that to create a TaskData object. The entity created before will never actually be stored in datastore and will only be used to create the id.

Reply:
I see.  It might be a good idea to update your [controller test](https://github.com/googleinterns/step171-2020/blob/master/src/test/java/com/rtb/projectmanagementtool/task/TaskControllerTest.java#L167) to test for this assumption.  The expectation would be that each of the returned tasks would have their own unique IDs, and doing a "get task by ID" operation would give you back the task with that unique ID.